pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/ApplyTableConfigCommand.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.tools.admin.command;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Splitter;
+import com.linkedin.pinot.common.config.CombinedConfig;
+import com.linkedin.pinot.common.config.CombinedConfigLoader;
+import com.linkedin.pinot.common.config.Serializer;
+import com.linkedin.pinot.controller.helix.ControllerRequestURLBuilder;
+import com.linkedin.pinot.tools.Command;
+import java.io.File;
+import java.io.InputStream;
+import java.util.List;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.kohsuke.args4j.Option;
+
+
+/**
+ * Command that applies a given table configuration to a cluster.
+ */
+public class ApplyTableConfigCommand extends AbstractBaseAdminCommand implements Command {
+
+  @Option(name = "-controllerUrl", required = true, metaVar = "<String>", usage = "Controller URL")
+  private String _controllerUrl;
+
+  @Option(name = "-tableConfigFile", required = true, metaVar = "<string>", usage = "Table configuration file to use.")
+  private String _tableConfigFile = null;
+
+  @Option(name = "-profile", required = false, metaVar = "<string>", usage = "Comma separated list of configuration profiles to use.")
+  private String _profileList = null;
+
+  @Option(name = "-showComputedConfig", required = false, usage = "Only display the computed configuration, without attempting to upload it")
+  private boolean _showComputedConfig = false;
+
+  @Option(name = "-help", required = false, help = true, aliases = { "-h", "--h", "--help" },
+      usage = "Print this message.")
+  private boolean _help = false;
+
+  @Override
+  public boolean execute() throws Exception {
```

Comment:
Does it make sense to separate out the logic to apply table config from the CLI? Doing so allows the former to be reusable (have found that to be useful in the past).

Reply:
After discussing this offline, we'll leave it this way since most of the logic (config profile handling, generation of the upload url) is already factored out. It's not exactly clear what would be the right API for people to reuse this, so we can extract a proper API at the time at which it would be needed.